### PR TITLE
feat(docs): nil plugin reference and howto

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----
CRAFT-2682

### CODE REVIEW

File: docs/common/craft-parts/how-to/plugins/nil_plugin.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+The following snippet makes use of `ref: override-build <part-properties-override-build>`
+to run a build and ensure that the built artifacts are staged. The following example uses
+`Tup`_ to build the ``hello`` target from the ``Tupfile`` provided in the ``source`` and
+later installed at the root of the part's installation directory to be ready for staging:
+
+.. code-block:: yaml
+
+   parts:
+     custom:
```
Comment (mr-cal) [inline docs/common/craft-parts/how-to/plugins/nil_plugin.rst:19]: Should `plugin: nil` be here?

File: docs/common/craft-parts/reference/plugins/nil_plugin.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Dependencies
+------------
+
+This plugin has no dependencies.
+
+
+How it works
+------------
+
+During the build step, this plugin essentially does nothing.
```
Comment (mr-cal) [inline docs/common/craft-parts/reference/plugins/nil_plugin.rst:33]: This raises more questions than it answers.  I would either remove the word "essentially" or explain what happens during the build step.

File: docs/common/craft-parts/how-to/plugins/nil_plugin.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Nil Plugin
+==========
+
+.. _how_to_unsupported_plugin:
+
+How to build with no plugin support
+-----------------------------------
+
+The following snippet makes use of `ref: override-build <part-properties-override-build>`
```
Comment (mr-cal) [inline docs/common/craft-parts/how-to/plugins/nil_plugin.rst:11]:
```suggestion
The following snippet makes use of :ref:`override-build <part-properties-override-build>`
```

**Review by mr-cal — APPROVED** [review]
Looks good, just some minor feedback.

File: docs/common/craft-parts/how-to/plugins/nil_plugin.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
+.. _how_to_nil_plugin:
```
Comment (tigarmo) [inline docs/common/craft-parts/how-to/plugins/nil_plugin.rst:1]: So I thought about this and I decided that I don't like this structure of the plugin examples being in how-tos. The example given here is more of an example of using an unsupported build tool to build a part, which is related to the `nil` plugin but then the plugin itself isn't the focus of the example. How-tos are also about being a series of steps/instructions towards doing something, like the how-to guide we have for [overriding a part's build](https://canonical-craft-parts.readthedocs-hosted.com/en/latest/common/craft-parts/how-to/override_build.html) (this example would be a better fit there than here, imo).

Finally, Diátaxis [likes examples in references](https://diataxis.fr/reference/#provide-examples) and the example that it gives (this is getting very meta) is "illustrating a command through an example of its usage" is pretty much our use-case for the plugins docs.

I'm happy to discuss this more if you want

**Review by tigarmo — CHANGES_REQUESTED** [review]

File: docs/common/craft-parts/reference/plugins/nil_plugin.rst (hunk 2 above)
Comment (sergiusens) [reply docs/common/craft-parts/reference/plugins/nil_plugin.rst]: fixed!

File: docs/common/craft-parts/how-to/plugins/nil_plugin.rst (hunk 3 above)
Comment (sergiusens) [reply docs/common/craft-parts/how-to/plugins/nil_plugin.rst]: this how to is now gone (for now)

File: docs/common/craft-parts/how-to/plugins/nil_plugin.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+Nil Plugin
```
[the remark this answers is not in the document]
Comment (sergiusens) [reply docs/common/craft-parts/how-to/plugins/nil_plugin.rst]: added

File: docs/common/craft-parts/how-to/plugins/nil_plugin.rst (hunk 4 above)
Comment (sergiusens) [reply docs/common/craft-parts/how-to/plugins/nil_plugin.rst]: I removed this how-to for now

File: docs/common/craft-parts/craft-parts.wordlist.txt [hunk 6]
```diff
@@ -226,6 +226,7 @@ Subtractive
 TarSource
 TimedWriter
 Tox
+Tup
```
Comment (tigarmo) [inline docs/common/craft-parts/craft-parts.wordlist.txt:229]: this one is not necessary anymore is it

**Review by tigarmo — APPROVED** [review]

File: docs/common/craft-parts/craft-parts.wordlist.txt (hunk 6 above)
Comment (sergiusens) [reply docs/common/craft-parts/craft-parts.wordlist.txt:229]: correcto!

**Pull request merged**